### PR TITLE
feat: Add domain locale detection (#1283)

### PR DESCRIPTION
**Pull request by cmaerz**
This implements basic domain locale detection now the `config.i18n.domains` section is respected.

I'm open for refactoring suggestions, as I'm not really deep in this project. Just wanted to get it in a pulll request, as I'm on vacation next week.

### CODE REVIEW

**Review by dphang — COMMENTED** [review]
Great, some unit tests would be good for this (can’t add to e2e tests yet since I don’t have a domain yet)

and also next we should support the domain locales (not just default locale)

File: packages/libs/core/src/route/locale.ts [hunk 1]
```diff
@@ -1,11 +1,37 @@
...
 export function addDefaultLocaleToPath(
   path: string,
-  routesManifest: RoutesManifest
+  routesManifest: RoutesManifest,
+  forceLocale: string | boolean = false
 ): string {
   if (routesManifest.i18n) {
-    const defaultLocale = routesManifest.i18n.defaultLocale;
+    const defaultLocale = forceLocale
+      ? forceLocale
```
Comment (dphang) [inline packages/libs/core/src/route/locale.ts:33]: Use ?? instead?

Comment (cmaerz) [reply packages/libs/core/src/route/locale.ts]: Changed this. Also rewrote some of that `addDefaultLocaleToPath` function.

**Comment by cmaerz** [comment]
I added some tests for the old `addDefaultLocaleToPath` function and the new `findDomainLocale` (bit hacky here, but tbh i had no better idea to mock that  `IncomingMessage` Type.

**Comment by dphang** [comment]
> I added some tests for the old `addDefaultLocaleToPath` function and the new `findDomainLocale` (bit hacky here, but tbh i had no better idea to mock that `IncomingMessage` Type).

Yeah I think that is ok for now, maybe I can try to see if there are some mocking libraries to help with it.

**Comment by dphang** [comment]
e2e tests passed: https://github.com/serverless-nextjs/serverless-next.js/actions/runs/972240984

**Review by dphang — APPROVED** [review]

**Pull request merged**